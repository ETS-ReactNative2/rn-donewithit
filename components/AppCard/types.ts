import {
  GestureResponderEvent,
  ImageSourcePropType,
  StyleProp,
  ViewStyle
} from 'react-native'

export type AppCardProps = {
  title: string
  image: ImageSourcePropType
  subTitle?: string
  style?: StyleProp<ViewStyle>
  onPress?: (e?: GestureResponderEvent) => void
}
