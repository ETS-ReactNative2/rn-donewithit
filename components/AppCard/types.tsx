import { ImageSourcePropType } from 'react-native'

export type AppCardProps = {
  title: string
  image: ImageSourcePropType
  subTitle?: string
}
