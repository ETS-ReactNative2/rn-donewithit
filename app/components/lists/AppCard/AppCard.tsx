import React, { useCallback } from 'react'
import {
  View,
  Image,
  StyleSheet,
  GestureResponderEvent,
  TouchableHighlight
} from 'react-native'
import COLORS from '../../../config/colors'
import CONSTS from './consts'
import AppText from '../../AppText'
import { AppCardProps } from './types'

function AppCard({
  title,
  subTitle,
  image,
  style,
  onPress,
  titleMaxLines = CONSTS.TITLE_MAX_LINES,
  subTitleMaxLines = CONSTS.SUBTITLE_MAX_LINES
}: AppCardProps) {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableHighlight
      underlayColor={COLORS.LIGHT_GRAY}
      activeOpacity={0.85}
      onPress={handlePress}
    >
      <View style={[style, styles.container]}>
        <Image style={styles.image} source={image} />
        <View style={styles.infoBox}>
          <AppText
            style={[styles.text, styles.title]}
            numberOfLines={titleMaxLines}
          >
            {title}
          </AppText>
          {subTitle && (
            <AppText
              style={[styles.text, styles.subTitle]}
              numberOfLines={subTitleMaxLines}
            >
              {subTitle}
            </AppText>
          )}
        </View>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    borderRadius: CONSTS.CONTAINER_BORDER_RADIUS,
    overflow: 'hidden'
  },
  image: {
    width: '100%',
    height: CONSTS.IMAGE_HEIGHT
  },
  infoBox: {
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL,
    paddingVertical: CONSTS.PADDING_VERTICAL
  },
  text: {
    fontSize: CONSTS.FONT_SIZE,
    lineHeight: CONSTS.LINE_HEIGHT,
    fontWeight: 'bold'
  },
  title: {
    color: COLORS.BLACK
  },
  subTitle: {
    color: COLORS.SECONDARY
  }
})

export default AppCard
