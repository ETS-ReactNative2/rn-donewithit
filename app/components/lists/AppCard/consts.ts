const CONSTS = {
  FONT_SIZE: 18,
  CONTAINER_BORDER_RADIUS: 10,
  PADDING_HORIZONTAL: 20,
  PADDING_VERTICAL: 15,
  LINE_HEIGHT: 32,
  IMAGE_HEIGHT: 200,
  TITLE_MAX_LINES: 1,
  SUBTITLE_MAX_LINES: 3
}

export default CONSTS
