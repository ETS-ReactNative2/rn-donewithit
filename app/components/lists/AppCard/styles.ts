import { StyleSheet } from 'react-native'
import COLORS from '../../../config/colors'
import CONSTS from './consts'

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    borderRadius: CONSTS.CONTAINER_BORDER_RADIUS,
    overflow: 'hidden'
  },
  image: {
    width: '100%',
    height: CONSTS.IMAGE_HEIGHT
  },
  infoBox: {
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL,
    paddingVertical: CONSTS.PADDING_VERTICAL
  },
  text: {
    fontSize: CONSTS.FONT_SIZE,
    lineHeight: CONSTS.LINE_HEIGHT,
    fontWeight: 'bold'
  },
  title: {
    color: COLORS.BLACK
  },
  subTitle: {
    color: COLORS.SECONDARY
  }
})

export default styles
