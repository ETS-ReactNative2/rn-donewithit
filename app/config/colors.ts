const COLORS = {
  PRIMARY: 'tomato',
  SECONDARY: 'turquoise',
  BACKGROUND_COLOR: '#f8f4f4',
  INPUT_BACKGROUND_COLOR: '#f9f2f2',
  DANGER: '#ff5252',
  GRAY: '#6e6969',
  LIGHT_GRAY: '#f8f4f4',
  WHITE: '#fff',
  BLACK: '#000',
  SILVER: '#C0C0C0',
  YELLOW: '#ffe66d'
}

export default COLORS
