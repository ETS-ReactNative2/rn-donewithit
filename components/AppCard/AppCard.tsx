
import React from 'react';
import { View, Image, Text, StyleSheet, ImageSourcePropType } from 'react-native';
import COLORS from '../../config/colors'
import CONSTS from './consts'

type AppCardProps = {
  title: string,
  subTitle?: string,
  image: ImageSourcePropType
}

function AppCard({
  title,
  subTitle,
  image,
}: AppCardProps) {
  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        source={image} />
      <View style={styles.infoBox}>
        <Text style={[styles.text, styles.title]}>{title}</Text>
        { subTitle && (<Text style={[styles.text, styles.subTitle]}>{subTitle}</Text>) }
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.WHITE,
    borderRadius: CONSTS.CONTAINER_BORDER_RADIUS,
    overflow: 'hidden',
  },
  image: {
    width: '100%',
    height: CONSTS.IMAGE_HEIGHT,
  },
  infoBox: {
    paddingHorizontal: CONSTS.PADDING_HORIZONTAL,
    paddingVertical: CONSTS.PADDING_VERTICAL,
  },
  text: {
    fontSize: CONSTS.FONT_SIZE,
    lineHeight: CONSTS.LINE_HEIGHT,
  },
  title: {
    color: COLORS.BLACK,
  },
  subTitle: {
    color: COLORS.SECONDARY
  }
})

export default AppCard;
