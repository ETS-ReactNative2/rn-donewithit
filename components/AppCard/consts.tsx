const CONSTS = {
  FONT_SIZE: 18,
  CONTAINER_BORDER_RADIUS: 10,
  PADDING_HORIZONTAL: 20,
  PADDING_VERTICAL: 15,
  LINE_HEIGHT: 32,
  IMAGE_HEIGHT: 200,
}

export default CONSTS
