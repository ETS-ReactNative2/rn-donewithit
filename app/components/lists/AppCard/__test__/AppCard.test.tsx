import React from 'react'
import { StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import AppCard from '../index'
import { COUCH } from '../../../../config/images'

const customStyles = StyleSheet.create({
  appCard: {
    marginHorizontal: 10,
    marginTop: 20
  }
})

const props = {
  style: customStyles.appCard,
  title: 'Couch in great condition',
  subTitle: '$1000',
  image: COUCH
}

describe('AppCard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppCard {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
