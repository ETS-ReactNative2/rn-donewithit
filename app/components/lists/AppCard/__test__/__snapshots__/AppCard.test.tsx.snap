// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        Object {
          "marginHorizontal": 10,
          "marginTop": 20,
        },
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "overflow": "hidden",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "paddingHorizontal": 20,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Courier",
              "fontSize": 13,
            },
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "lineHeight": 32,
              },
              Object {
                "color": "#000",
              },
            ],
          ]
        }
      >
        Couch in great condition
      </Text>
      <Text
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "Courier",
              "fontSize": 13,
            },
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "lineHeight": 32,
              },
              Object {
                "color": "turquoise",
              },
            ],
          ]
        }
      >
        $1000
      </Text>
    </View>
  </View>
</View>
`;
